File: data_structures/trees/golang/implementation/create_basic.go
package main

// BinarySearchTreeNode Class
type BinarySearchTreeNode struct {
	value int
	left  *BinarySearchTreeNode
	right *BinarySearchTreeNode
}

func main() {
	/*
	       1
	      / \
	     2   3
	    / \
	   4   5
	*/

	root := &BinarySearchTreeNode{1, nil, nil}
	root.left = &BinarySearchTreeNode{2, nil, nil}
	root.right = &BinarySearchTreeNode{3, nil, nil}
	root.left.left = &BinarySearchTreeNode{4, nil, nil}
	root.left.right = &BinarySearchTreeNode{5, nil, nil}

}

File: data_structures/trees/cpp/implementation/create_basic.cpp
#include <iostream>
using namespace std;

struct BinarySearchTreeNode
{
    int value;
    struct BinarySearchTreeNode *left, *right;
    BinarySearchTreeNode(int value)
    {
        this->value = value;
        left = right = NULL;
    }
};

int main()
{
    /*
	       1
	      / \
	     2   3
	    / \
	   4   5
	*/

    struct BinarySearchTreeNode *root = new BinarySearchTreeNode(1);
    root->left             = new BinarySearchTreeNode(2); 
    root->right         = new BinarySearchTreeNode(3); 
    root->left->left     = new BinarySearchTreeNode(4); 
    root->left->right = new BinarySearchTreeNode(5);  

    cout << "Created BST! "<<endl;

    return 0;
}